### DISCUSSION TITLE
Kafka Companion clearing the topics for the component tests

### DISCUSSION THREAD

**Discussion by cj19** [smallrye/smallrye-reactive-messaging#2662] [Q&A] [answered]
I am using Kafka Companion for test purposes and I am looking for a solution to clear/purge the topics. The problem is that if I have multiple test cases, then there is a huge change to exhcange the messages between the test cases, which makes the tests fail. It means that some of the consumers pick the wrong message out of the topic, because the tests use the same input and output topics. 
Which is the best solution for this?
I am using Quarkus framework with gradle and java 17.

**Comment by ozangunalp** [comment, marked as answer]
You can either delete and recreate the topic. Or call `companion.topics().clear(topicNames)` to delete records. 
Note that deleting records inside a topic doesn't delete committed offsets.

**Reply by cj19** [reply]
> You can either delete and recreate the topic. Or call `companion.topics().clear(topicNames)` to delete records. Note that deleting records inside a topic doesn't delete committed offsets.

Well, turned out for my tests the clearing or deleting a topic is not an option.
I have multiple components and some of them use the same topics etc. When i run all the tests for all of the components it can cause "interference" between the tests. I track the offsets for the topics in hashmaps, but still sometimes it can happen that the offset is not right or not the preferred message gets picked out of the topic. Any suggestions how to solve this? @ozangunalp
